test/metabase/lib/remove_replace_test.cljc

```diff
@@ -377,7 +378,32 @@
                   (as-> <> (lib/expression <> "expr" (lib/aggregation-ref <> 0)))
                   (lib/append-stage)
                   (lib/filter (lib/= [:field {:lib/uuid (str (random-uuid)) :base-type :type/Integer} "sum"] 1))
-                  (lib/replace-clause 0 (first aggregations) (lib/sum (meta/field-metadata :venues :price)))))))))
+                  (lib/replace-clause 0 (first aggregations) (lib/sum (meta/field-metadata :venues :price))))))))
+  (testing "replacing with metric should work"
+    (let [metadata-provider (lib.tu/mock-metadata-provider
+                              {:database meta/metadata
+                               :tables   [(meta/table-metadata :venues)]
+                               :fields   [(meta/field-metadata :venues :price)]
+                               :metrics  [{:id          100
+                                           :name        "Sum of Cans"
+                                           :table-id    (meta/id :venues)
+                                           :definition  {:source-table (meta/id :venues)
+                                                         :aggregation  [[:sum [:field (meta/id :venues :price) nil]]]
+                                                         :filter       [:= [:field (meta/id :venues :price) nil] 4]}
+                                           :description "Number of toucans plus number of pelicans"}]})
```

Comment:
It probably wouldn't be a bad idea to just make this a predefined metadata provider in `lib.tu` so we can reuse it whenever we need to test with metrics in the future